k8s/core/pods.go

```diff
@@ -69,6 +69,8 @@ type PodOps interface {
 	WaitForPodDeletion(uid types.UID, namespace string, timeout time.Duration) error
 	// RunCommandInPod runs given command in the given pod
 	RunCommandInPod(cmds []string, podName, containerName, namespace string) (string, error)
+	// RunCommandInPodEx is extended version of RunCommandInPod
+	RunCommandInPodEx(cmds []string, podName, containerName, namespace string, useTTY bool, stdin io.Reader, stdout, stderr io.Writer) error
```

Comment:
I see we are adding additional parameters here and changing the interface function name.

When we update this to add a new parameter, it'll have to yet another new name to avoid breaking changes.

A better approach for this would be to use request object - i.e. 
```
type RunCommandInPodRequest struct {
  Cmds []string
  podName string
  ...
  stdout io.Writer
  stderr io.Writer
}
```

Reply:
> I see we are adding additional parameters here and changing the interface function name.

Yes -- the original `RunCommandInPod()` did not take the STDIN param, also did not separate STDOUT and STDERR streams.  The "extended" `RunCommandInPodEx()` does all this.

> A better approach for this would be to use request object - i.e.

So.. you're just asking to change the 8 input parameters, and take only the 1 param (the `RunCommandInPodRequest struct`)?   Note that this will "break" the uniformity of the API  (8 input params is "a lot", good point there, but no other POD-function takes struct as an input param).

Reply:
I would suggest one new function RunCommandInPodEx with a request object `RunCommandInPodExRequest`.

That way, you can add parameters without changing the function signature every time. Makes it more maintainable

Reply:
Ok.. change implemented in [dfcfbc0](https://github.com/portworx/sched-ops/pull/302/commits/dfcfbc024ba7381a445de458588063f067de59a8)